pages/[network]/[chain]/pull-request.tsx

```diff
@@ -186,48 +258,13 @@ export default function PullRequestPage() {
   }
 
   useEffect(() => {
-    if (!state.currentBounty?.data ||
-        !prId ||
-        state?.spinners?.pullRequests) return;
-
-    if (pullRequest) {
-      const currentPr = state.currentBounty?.data?.pullRequests?.find((pr) => +pr.githubId === +prId);
-
-      if (pullRequest.isReady !== currentPr.isReady || pullRequest.status !== currentPr.status)
-        setPullRequest(previous => ({
-          ...previous,
-          isReady: currentPr.isReady,
-          status: currentPr.status
-        }));
-
-      return;
-    }
-
-    dispatch(changeLoadState(true));
-    dispatch(changeSpinners.update({pullRequests: true}));
-
-    getExtendedPullRequestsForCurrentBounty()
-      .then(pullRequests => {
-        dispatch(changeCurrentBountyData(Object.assign(state.currentBounty.data, { pullRequests })));
-
-        return pullRequests.find((pr) => +pr.githubId === +prId);
-      })
-      .then(setPullRequest)
-      .finally(() => {
-        dispatch(changeLoadState(false))
-        dispatch(changeSpinners.update({pullRequests: false}));
-      });
-
-  }, [state.currentBounty?.data, prId]);
-
-  useEffect(() => {
-    if (review && pullRequest && state.currentUser?.login && state.connectedChain?.matchWithNetworkChain)
+    if (review && currentPullRequest && state.currentUser?.login && state.connectedChain?.matchWithNetworkChain)
       setShowModal(true);
-  }, [review, pullRequest, state.currentUser, state.connectedChain?.matchWithNetworkChain]);
+  }, [review, currentPullRequest, state.currentUser, state.connectedChain?.matchWithNetworkChain]);
```

Comment:
currentPullRequest will always exist, and if it doesn't we should have shown a 404 and that should happen via https://nextjs.org/docs/pages/api-reference/functions/get-server-side-props#notfound

it's also impossible to see the "make review" or any button before actually being connected and on the matchingNetwork meaning that showModal can have a default value on creation via [showModal, setShowModal] = useState(!!query.review)

in essence, this useEffect is not needed ;)